kinode/packages/app_store/app_store/src/state.rs

```diff
@@ -0,0 +1,566 @@
+        self.update_downloaded_package(package_id, |package_state| {
+            package_state.mirroring = false;
+        })
+    }
+
+    pub fn start_auto_update(&mut self, package_id: &PackageId) -> bool {
+        self.update_downloaded_package(package_id, |package_state| {
+            package_state.auto_update = true;
+        })
+    }
+
+    pub fn stop_auto_update(&mut self, package_id: &PackageId) -> bool {
+        self.update_downloaded_package(package_id, |package_state| {
+            package_state.auto_update = false;
+        })
+    }
+
+    /// saves state
+    pub fn populate_packages_from_filesystem(&mut self) -> anyhow::Result<()> {
+        let Message::Response { body, .. } = Request::to(("our", "vfs", "distro", "sys"))
+            .body(serde_json::to_vec(&vfs::VfsRequest {
+                path: "/".to_string(),
+                action: vfs::VfsAction::ReadDir,
+            })?)
+            .send_and_await_response(VFS_TIMEOUT)??
+        else {
+            return Err(anyhow::anyhow!("vfs: bad response"));
+        };
+        let response = serde_json::from_slice::<vfs::VfsResponse>(&body)?;
+        let vfs::VfsResponse::ReadDir(entries) = response else {
+            return Err(anyhow::anyhow!("vfs: unexpected response: {:?}", response));
+        };
+        for entry in entries {
+            // ignore non-package dirs
+            let Ok(package_id) = entry.path.parse::<PackageId>() else {
+                continue;
+            };
+            if entry.file_type == vfs::FileType::Directory {
+                let zip_file = vfs::File {
+                    path: format!("/{}/pkg/{}.zip", package_id, package_id),
+                    timeout: 5,
+                };
+                let Ok(zip_file_bytes) = zip_file.read() else {
+                    continue;
+                };
+                // generate entry from this data
+                // for the version hash, take the SHA-256 hash of the zip file
+                let our_version = utils::generate_version_hash(&zip_file_bytes);
+                let manifest_file = vfs::File {
+                    path: format!("/{}/pkg/manifest.json", package_id),
+                    timeout: 5,
+                };
+                let manifest_bytes = manifest_file.read()?;
+                // the user will need to turn mirroring and auto-update back on if they
+                // have to reset the state of their app store for some reason. the apps
+                // themselves will remain on disk unless explicitly deleted.
+                self.add_downloaded_package(
+                    &package_id,
+                    PackageState {
+                        mirrored_from: None,
+                        our_version,
+                        installed: true,
+                        verified: true,      // implicity verified
+                        caps_approved: true, // since it's already installed this must be true
```

Comment:
> since it's already installed this must be true

Is this true? What if I
1. Download a package from `foo.os`
2. Do not install
3. Turn off node
4. Turn on node

Reply:
oof, yeah. problematique